rust/cubesql/cubesql/src/compile/rewrite/rules/members.rs

```diff
@@ -1478,6 +1511,78 @@ impl MemberRules {
             )
     }
 
+    fn select_distinct_dimensions(
+        &self,
+        alias_to_cube_var: &'static str,
+        members_var: &'static str,
+        filters_var: &'static str,
+        left_ungrouped_var: &'static str,
+    ) -> impl Fn(&mut CubeEGraph, &mut Subst) -> bool {
+        let alias_to_cube_var = var!(alias_to_cube_var);
+        let members_var = var!(members_var);
+        let filters_var = var!(filters_var);
+        let left_ungrouped_var = var!(left_ungrouped_var);
+        let meta_context = self.meta_context.clone();
+
+        move |egraph, subst| {
+            let empty_filters = &egraph
+                .index(subst[filters_var])
+                .data
+                .is_empty_list
+                .unwrap_or(true);
+            let ungrouped = var_iter!(egraph[subst[left_ungrouped_var]], CubeScanUngrouped)
+                .into_iter()
+                .any(|v| *v);
+
+            if !empty_filters && ungrouped {
+                return false;
+            }
+
+            let res = match egraph
+                .index(subst[members_var])
+                .data
+                .member_name_to_expr
+                .as_ref()
+            {
+                Some(names_to_expr) => {
+                    names_to_expr.list.iter().all(|(_, member, _)| {
+                        // we should allow transform for queries with dimensions only,
+                        // as it doesn't make sense for measures
+                        match member {
+                            Member::Dimension { .. } => true,
+                            Member::VirtualField { .. } => true,
+                            Member::LiteralMember { .. } => true,
+                            _ => false,
+                        }
+                    })
+                }
+                None => {
+                    // this might be the case of `SELECT DISTINCT *`
+                    // we need to check that there are only dimensions defined in the referenced cube
+                    let aliases_to_cube: Vec<_> =
+                        var_iter!(egraph[subst[alias_to_cube_var]], CubeScanAliasToCube)
+                            .cloned()
+                            .collect();
+
+                    if let Some(cube_name) = aliases_to_cube
+                        .first()
+                        .and_then(|f| f.first().map(|(_, c)| c))
```

Comment:
`f.first()` does not look correct here
If `CubeScan` is scanning from tow cubes (because of join on `__cubeJoinField`) it would have two cubes in `CubeScanAliasToCube`, and this rule have to check both for being `dimension-only`